docs/Configuration.md

```diff
@@ -68,24 +74,36 @@ Available keys : `host`, `user`, `password`, `database`.
 
 ### host
 
+An [LDH domain name].
```

Comment:
Shouldn't we also allow IP address ?

Reply:
After looking at the unit tests, and again at the regular expression, it looks like IP address are not rejected here. However, we might need to be clearer on the fact that we accept them, because the provided link is about restriction on domain names and doesn't say anything on IP address.

Reply:
IPv6 address is rejected, isn't it?

Reply:
I'll update to "An LDH domain name or and IP address".

Reply:
A note on how IPv6 should be formatted for MySQL :

> When connecting to a MySQL Server with IPv6, a bracketed IPv6 address should be used.

https://metacpan.org/pod/DBD::mysql#connect

(sorry to make this PR heavier with my remarks :/ )
(and reading the PR title again, this is about validating simple properties, if you feel this isn't simple anymore, feel free to say so)

Reply:
At some point I was thinking about splitting this PR into controversial and non-controversial configuration properties so we could have the non-controversial ones merged and get a fresh PR for the controversial ones. But now it feels like we're getting close to the finish line.

Also from the start I actually expected this PR to be a rough ride because it contains so many details. And one of the worries I've been having since is that it would start forbidding reasonable edge cases. This IPv6 thing is a prime example of that. In hindsight I guess it was unwise cram all of these properties into a single PR.

I'll make the MySQL adapter bracket IPv6 addresses and see if I can get MariaDB to listen to IPv6.

Reply:
I just realized this is a breaking change. If you wanted IPv6 before you'd specify `host = [2001:db8::1]`, but after this PR you'll have update your config to `host = 2001:db8::1`.

Reply:
I made the update and afterwards I was able to connect to MariaDB over IPv6. However I had to cheat - the IPv6 regex doesn't match `::1` so I had to hack around that.